src/python_testing/mdns_discovery/data_clases/mdns_service_info.py

```diff
@@ -0,0 +1,76 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional
+
+from zeroconf.asyncio import AsyncServiceInfo
+
+from .json_serializable import JsonSerializable
+
+
+@dataclass
+class MdnsServiceInfo(JsonSerializable):
+    service_info: AsyncServiceInfo = field(repr=False, compare=False)
+
+    service_name: Optional[str] = field(init=False)
+    service_type: Optional[str] = field(init=False)
+    instance_name: Optional[str] = field(init=False)
+    hostname: Optional[str] = field(init=False)
+    port: Optional[int] = field(init=False)
+    addresses: Optional[List[str]] = field(init=False)
+    txt: Optional[Dict[str, str]] = field(init=False)
+    priority: Optional[int] = field(init=False)
+    interface_index: Optional[int] = field(init=False)
+    weight: Optional[int] = field(init=False)
+    host_ttl: Optional[int] = field(init=False)
+    other_ttl: Optional[int] = field(init=False)
+
+    def __post_init__(self):
+        si = self.service_info
+
+        self.service_name = si.name
```

Comment:
why do we do this split? why are we not doing:

```py
@property
def service_type(self):
  return self.si.type
```

or overall why have these methods at all? why not let the caller do `foo.si.type` instead of `foo.service_type` ? maybe si should be named `service_info`. Why do we denormalize/flatten the data? this is a lot of repeated code just to hoist members one level up.

Reply:
This is a brand new class added by this PR. The fix is not difficult. Why are we moving it as a separate issue?

Reply:
I'm going to resolve this so the code can merge - appears this was updated.